server/blob/blob-cassandra/src/test/java/org/apache/james/blob/cassandra/cache/CachedBlobStoreTest.java

```diff
@@ -274,4 +281,52 @@ public void shouldNotCacheWhenReadWithBigStream() {
             assertThat(Mono.from(cache.read(blobId)).blockOptional()).isEmpty();
         });
     }
+
+    @Test
+    public void readBlobStoreCacheShouldPublishTimerMetrics() {
+        BlobId blobId = Mono.from(backend.save(DEFAULT_BUCKETNAME, APPROXIMATELY_FIVE_KILOBYTES, SIZE_BASED)).block();
+
+        testee.read(DEFAULT_BUCKETNAME, blobId);
+        testee.read(DEFAULT_BUCKETNAME, blobId);
+
+        assertThat(metricFactory.executionTimesFor(BLOBSTORE_CACHED_LATENCY_METRIC_NAME))
+            .hasSize(2);
+    }
+
+    @Test
+    public void readBlobStoreCacheShouldCountWhenHit() {
+        BlobId blobId = Mono.from(testee.save(DEFAULT_BUCKETNAME, APPROXIMATELY_FIVE_KILOBYTES, SIZE_BASED)).block();
+
+        testee.read(DEFAULT_BUCKETNAME, blobId);
+        testee.read(DEFAULT_BUCKETNAME, blobId);
+
+        assertThat(metricFactory.countFor(BLOBSTORE_CACHED_HIT_COUNT_METRIC_NAME))
+            .isEqualTo(2);
+    }
+
+    @Test
+    public void readBlobStoreCacheShouldCountWhenMissed() {
+        BlobId blobId = Mono.from(backend.save(DEFAULT_BUCKETNAME, APPROXIMATELY_FIVE_KILOBYTES, SIZE_BASED)).block();
+        cache.remove(blobId);
+
+        testee.read(DEFAULT_BUCKETNAME, blobId);
+
+        assertThat(metricFactory.countFor(BLOBSTORE_CACHED_MISS_COUNT_METRIC_NAME))
+            .isEqualTo(1);
+    }
+
+    @Test
+    public void metricShouldNotWorkWhenReadNonExistingBlob() {
+        SoftAssertions.assertSoftly(soflty -> {
+            assertThatThrownBy(() -> testee.read(DEFAULT_BUCKETNAME, new TestBlobId.Factory().randomId()))
+                .isInstanceOf(ObjectNotFoundException.class);
+
+            assertThat(metricFactory.countFor(BLOBSTORE_CACHED_MISS_COUNT_METRIC_NAME))
+            .isEqualTo(0);
+            assertThat(metricFactory.countFor(BLOBSTORE_CACHED_HIT_COUNT_METRIC_NAME))
+            .isEqualTo(0);
+            assertThat(metricFactory.executionTimesFor(BLOBSTORE_CACHED_LATENCY_METRIC_NAME))
+            .hasSize(0);
```

Comment:
IMO it still makes sense to measure time it takes to know "that we have nothing", I think timing still needs to be recorded.

Reply:
Use a Soft assertion at least here.